### ISSUE TITLE
Allow subclassing of Selenium2Driver class

### ISSUE THREAD

**Issue by gggeek** [minkphp/MinkSelenium2Driver#231]
Currently I have a problem with subclassing the Selenium2 driver (I created a version which adds support for extra SauceLabs features): the method `setWebDriver()` is typehinted to WebDriver, which is a final class.
I am passing in my own version of a WebDriver, which extends `AbstractWebDriver`, and to make it work have reimplemented the `setWebDriver` method in my subclass of `Selenium2Driver`.
But the fact that the reimplementation removes the typehinting generates a php strict error, which can make the test fail depending on php configuration.

In short: I think that setWebDriver method should use a 'wider' typehinting

**Comment by gggeek** [comment]
PS: Maybe this could be fixed in the webdriver repo... creating an issue there

**Comment by stof** [comment]
We cannot typehint AbstractWebDriver, as we rely on methods not available in it.

**Comment by gggeek** [comment]
Exact. That's why I proposed this: https://github.com/instaclick/php-webdriver/pull/72

**Comment by stof** [comment]
@gggeek what kind of features have you added ?

**Comment by gggeek** [comment]
The WebDriverInterface, which sports the only method I saw used in the Selenium2Driver: `session()`

**Comment by stof** [comment]
What I'm asking is what kind of features you are adding in your subclass.

**Comment by gggeek** [comment]
Sorry, I realized that after answering.

Support for a proxy

**Comment by aik099** [comment]
I've commended on https://github.com/instaclick/php-webdriver/issues/71.

I think, that for now ability to specify curl options for every curl call made in WebDriver, that later you specify upon Mink driver instantiation would be what you're after.
